hpx/components/containers/container_distribution_policy.hpp

```diff
@@ -28,14 +30,18 @@ namespace hpx
     struct container_distribution_policy
       : components::default_distribution_policy
     {
+    private:
+        typedef components::default_distribution_policy base_type;
+
     public:
         container_distribution_policy()
           : num_partitions_(std::size_t(-1))
         {}
 
         container_distribution_policy operator()(std::size_t num_partitions) const
         {
-            return container_distribution_policy(num_partitions, localities_);
+            auto localities = this->base_type::ensure_localities();
```

Comment:
do we really need to assign this to a temporary and introduce reference counting here?

Reply:
YEs we need to do that because of the asynchronous functionality here: https://github.com/STEllAR-GROUP/hpx/pull/3087/files#diff-61639ba7144f3b99ed04835f11dd7924R163.

Reply:
Sure, this particular function however doesn't need it, it could directly pass on the `*localities_`

Reply:
`localities_` might be empty (or even invalid), while using `ensure_localities()` ensures using a properly initialized vector of localities.